File: app-backend/src/main/java/dev/andrei/app_backend/dto/review/MyReviewDto.java
package dev.andrei.app_backend.dto.review;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

/** A review as shown on the author's own "My reviews" list (carries the location it belongs to). */
public record MyReviewDto(
        UUID id,
        UUID locationId,
        String locationName,
        String content,
        Instant createdAt,
        double overallScore,
        List<AttributeScoreDto> attributeScores
) {}

File: appfrontend/app/src/main/java/dev/andrei/app_frontend/data/remote/dto/MyReviewDto.kt
package dev.andrei.app_frontend.data.remote.dto

// A review by the current user, carrying the location it belongs to.
data class MyReviewDto(
    val id: String,
    val locationId: String,
    val locationName: String,
    val content: String,
    val createdAt: String,
    val overallScore: Float,
    val attributeScores: List<AttributeScoreDto>
)